### PR TITLE
FIX Performance issues when a blog has a large number of tags.

### PR DESCRIPTION
**Pull request by micmania1**

### CODE REVIEW

File: code/admin/GridFieldMergeAction.php [hunk 1]
```diff
@@ -69,27 +89,48 @@ public function getColumnsHandled($gridField)
...
-
-            return $dropdown->Field() . $action->Field() . '<a title="Move posts to" class="MergeActionReveal">move posts to</a>';
+        if ($columnName === 'MergeAction') {
+
+            if(!$record->hasMethod($this->childMethod) || $record->{$this->childMethod}()->count() < 1) {
+                return '';
+            }
+
+            $children = $record->{$this->childMethod}();
+            $cacheKey = md5($children->dataQuery()->sql());
```
Comment (tractorcow) [inline code/admin/GridFieldMergeAction.php]: Won't work if two different queries have the same SQL, but different parameters. :)

Comment (tractorcow) [inline code/admin/GridFieldMergeAction.php]: What parameters do you need... $record->ID and $this->childMethod right? That might be enough.

Comment (tractorcow) [inline code/admin/GridFieldMergeAction.php]: Please update cacheKey to include hashed params. You can get them passed by by ref from the sql($args) method.

**Comment by dhensby** [comment]
Bump

**Comment by micmania1** [comment]
This didn't help so i'm going to close. I'll follow up on the original ticket #350

**Pull request closed without merging**